core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -165,12 +124,141 @@ object ConsumerGroupCommand extends Logging {
                                                 consumerId: Option[String], host: Option[String],
                                                 clientId: Option[String], logEndOffset: Option[Long])
 
+  protected case class MemberAssignmentState(group: String, consumerId: Option[String], host: Option[String],
```

Comment:
Having so many `Option` fields seems less than ideal unless every permutation actually needs to be represented. Maybe it could be simplified by encapsulating the optional state into a single case class? For example:
```scala
case class AssignmentState(consumerId: String, host: String, clientId: String, numPartitions: Int, assignment: List[TopicPartition])
case class MemberAssignmentState(group: String, assignment: Option[AssignmentState])
```
Same for `GroupState` below.